.gitattributes

```diff
@@ -6,32 +6,15 @@
 
 # Normalize EOL for all files that Git considers text files.
 * text=auto eol=lf
+
 # The above only works properly for Git 2.10+, so for older versions
 # we need to manually list the binary files we don't want modified.
 *.mp3 binary
 *.png binary
 
-# Files to exclude from asset-lib download.
-/.github export-ignore
-/.reuse export-ignore
-/LICENSES export-ignore
-/assets export-ignore
-/addons/gd-plug export-ignore
-/docs export-ignore
-/misc export-ignore
-/test export-ignore
-/.gitattributes export-ignore
-/.gitignore export-ignore
-/.gut_editor_config.json export-ignore
-/.gutconfig.json export-ignore
-/.gutconfig_all.json export-ignore
-/CREDITS.md export-ignore
-/LICENSE export-ignore
-/README.md export-ignore
-/default_env.tres export-ignore
-/icon.png export-ignore
-/icon.png.import export-ignore
-/plug.gd export-ignore
-/project.godot export-ignore
-/requirements.txt export-ignore
-/*.license export-ignore
+# Exclude all top-level files and directories (except addons) from zip downloads.
+# This makes installing through the AssetLib easier, because no files and folders
+# need to be unchecked.
+/**				        export-ignore
+/addons/glam			!export-ignore
+/addons/glam/**			!export-ignore
```

Comment:
**blocker:** This is nice, but unfortunately doesn't work. It results in an empty zip archive. There is currently a check in the workflow for this which fails with this `.gitattributes` file.
https://github.com/lihop/glam/blob/43a9fbe3baf281175b34d8121fdbc522e4e44026/.github/workflows/main.yml#L78-L115
Adding files and forgetting to update `.gitattributes` has caught me out a lot. It would be nice if there were a pre-commit hook for it. I'm considering adding one to setup-godot as I would also use it in other projects.